### DISCUSSION TITLE
login help after server update.

### DISCUSSION THREAD

**Discussion by rpyne** [postfixadmin/postfixadmin#727] [Q&A]
After a recent update to my server, I can no longer log in to postfixadmin version 3.2 on an installation I've been running for several years. The login page just refreshes and the log shows a success (200). I tried installing version 3.3.13, and it gives a server 500 with nothing in the error log.

Any tips or help in solving the issue will be greatly appreciated.

**Comment by DavidGoodwin** [comment]
Are you sure there's nothing in the PHP or Apache error log?

Does the setup.php page show anything?

**Reply by rpyne** [reply]
Okay, I got it to display errors. I forgot that I needed to restart FPM as well as the webserver after changing php.ini.

The messages showing now are:

-----------------------------------------------------------------
Warning: require_once(/var/www/postfixadmin-3.3.13/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/postfixadmin-3.3.13/common.php on line 3

Fatal error: require_once(): Failed opening required '/var/www/postfixadmin-3.3.13/vendor/autoload.php' (include_path='.:/usr/share/pear:/usr/share/php') in /var/www/postfixadmin-3.3.13/common.php on line 3
-------------------------------------------------------------------

Which is strange to me because common.php is in /var/www/postfixadmin-3.3.13/public, which is also my DocumentRoot.

**Comment by rpyne** [comment]
Doing more digging on why version 3.2 won't let me log in now, my mariadb error log is saying 

[Warning] Aborted connection 9491 to db: 'postfixadmin' user: 'postfixadmin' host: 'localhost' (Got an error reading communication packets)

This is new since updating the server from CentOS 8 to Rocky Linux 8. Prior to the update, it was working fine.

**Comment by DavidGoodwin** [comment]
Run the 'install.sh' shell script - that should fix the autoload.php stuff.

**Comment by rpyne** [comment]
@DavidGoodwin Thank you!

That got setup.php to run. I was able to fix the ssl access problems using the info in https://github.com/postfixadmin/postfixadmin/issues/381#issuecomment-678269198

Still working on solving 

⛔Password Hashing - attempted to use configured encrypt backend (dovecot:BLF-CRYPT) triggered an error: /usr/bin/doveadm pw -r 12 failed, see error log for details
⛔Possibly helpful error_log messages -
[02-May-2023 08:32:31 America/Boise] Failed to read password from /usr/bin/doveadm pw -r 12 ... stderr: Error: net_connect_unix(/run/dovecot/stats-writer) failed: Permission denied
, password: {BLF-CRYPT}$2y$12$wYFdwJw1VH1kNbBZuCl05OXQ5g/SMX91Ftgf1fw6qTNOtXCPvjPqy

**Comment by DavidGoodwin** [comment]
You'll need to restart apache/the webserver after you change the www-data group membership. The easiest way would be to reboot the server.

**Reply by rpyne** [reply]
Thanks. setting  664 permissions on the socket and restarting got me through the setup.php and can now login. main.php is displaying

Fatal error: Uncaught --> Smarty Compiler: Syntax error in template "file:/var/www/postfixadmin-3.3.13/templates/menu.tpl" on line 24 "<a class="navbar-brand" href='main.php'><img id="login_header_logo" src="{$CONF.theme_logo | default('images/postbox.png')}" alt="Logo"/></a>" - Unexpected "|" <-- thrown in /var/www/postfixadmin-3.3.13/templates/menu.tpl on line 24

**Reply by rpyne** [reply]
It appears that commenting out line 24 and uncommenting line 23 makes everything work.

**Comment by DavidGoodwin** [comment]
Thanks ! see 702bb6cdc30dbf3005891f10aa2a4962929c23eb - that's me mixing up twig syntax with smarty's.
